pkg/controllers/leaderworkerset_controller.go

```diff
@@ -511,19 +515,21 @@ func (r *LeaderWorkerSetReconciler) updateStatus(ctx context.Context, lws *leade
 	return updateDone, nil
 }
 
-// iterateReplicas will iterate the leader pods together with corresponding worker statefulsets
-// to check the replica state, and return two values and an error in the end:
-//   - The first value represents the number of continuous ready replicas ranging from the last index to 0,
-//     to help us judge whether we can update the Partition or not.
-//   - The second value represents the unready replicas whose index is smaller than leaderWorkerSet Replicas.
-func (r *LeaderWorkerSetReconciler) iterateReplicas(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet, stsReplicas int32, revisionKey string) (int32, int32, error) {
+type replicaState struct {
+	ready   bool
+	updated bool
```

Comment:
Seems no need to distinguish ready and updated separately.

Reply:
calculateLWSUnreadyReplicas/calculateContinuousReadyReplicas could use a joint `ready&updated` state while partitionAccountingForUnavailable uses `ready` only